File: ControlCenter.Infrastructure/Prerequisite.fs
namespace ControlCenter.Infrastructure

module Prerequisite =
    open System
    open System.IO
    open System.Resources
    open System.Reflection
    open System.Globalization
    open System.Collections
    open System.Linq
    open System.Text

    let internal prerequisites = ["GTASACarPics"; "GTASACars"; "GTASACarTypes"; "GTASACheats"; "GTASAColors"; "GTASALocations"; "GTASAShortcuts"; "GTASAWeapons"]

    /// Get specific resources from resource
    let internal GetResource =
        let rm = ResourceManager("Resources", Assembly.GetExecutingAssembly())
        let resourceSet = rm.GetResourceSet(CultureInfo.CurrentUICulture, true, true) :> IEnumerable
        fun name -> resourceSet.Cast<DictionaryEntry>().FirstOrDefault(fun r -> (string)r.Key = name).Value

    /// Check whether GTASAConsole.ini exists, if not generate it from resources;
    let internal RegenerateXml name =
        let xml = GetResource name :?> string
        File.WriteAllText(@".\" + name + ".xml", xml)
    
    /// Check prerequisites and generate missing prerequisites
    let Check() =
        prerequisites
            |> List.iter (fun p ->
                let _sb = new StringBuilder()
                let name = _sb.Append(AppDomain.CurrentDomain.BaseDirectory).Append(p).Append(".xml").ToString()
                if not (File.Exists(name)) then RegenerateXml p)

File: src/ControlCenter.Infrastructure/Prerequisite.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Resources;
using System.Text;

namespace ControlCenter.Infrastructure
{
    public static class Prerequisite
    {
        private static readonly IList<string> Prerequisites = new List<string>
        {
            "GTASACarPics",
            "GTASACars",
            "GTASACarTypes",
            "GTASACheats",
            "GTASAColors",
            "GTASALocations",
            "GTASAShortcuts",
            "GTASAWeapons"
        };

        private static object GetResource(string name)
        {
            var rm = new ResourceManager("ControlCenter.Infrastructure.Properties.Resources", Assembly.GetExecutingAssembly());
            var resourceSet = rm.GetResourceSet(CultureInfo.CurrentCulture, true, true);
            return resourceSet.Cast<DictionaryEntry>().FirstOrDefault(r => (string) r.Key == name).Value;
        }

        private static void RegenerateXml(string name)
        {
            var xml = GetResource(name) as string;
            File.WriteAllText(@".\" + name + ".xml", xml);
        }

        /// <summary>
        /// Check ControlCenter's prerequisites, if not exist regenerate files
        /// </summary>
        public static void Check()
        {
            foreach (var prerequisite in Prerequisites)
            {
                var sb = new StringBuilder();
                var name = sb.Append(AppDomain.CurrentDomain.BaseDirectory).Append(prerequisite).Append(".xml")
                    .ToString();
                if (!File.Exists(name))
                {
                    RegenerateXml(prerequisite);
                }
            }
        }
    }
}